awxkit/awxkit/cli/resource.py

```diff
@@ -122,6 +175,188 @@ def handle(self, client, parser):
+            if page is None:
+                page = endpoint.post(post_data)
+            else:
+                page = page.put(post_data)
+
+            self._natural_key[freeze(get_natural_key(page))] = page
+
+    def register_existing_assets(self, resource):
+        endpoint = getattr(self.v2, resource)
+        options = endpoint.options().json['actions']['POST']
+        self._options[resource] = options
+
+        results = endpoint.get(all_pages=True).results
+        for asset in results:
+            self._natural_key[freeze(get_natural_key(asset))] = asset
+
+    def assign_roles(self, page, roles):
+        role_endpoint = page.json['related']['roles']
+        for role in roles:
+            if 'content_object' not in role:
+                continue  # admin role
+            obj_page = self.get_by_natural_key(role['content_object'])
+            if obj_page is not None:
+                role_page = obj_page.get_object_role(role['name'], by_name=True)
+                try:
+                    role_endpoint.post({'id': role_page['id']})
+                except exc.NoContent:  # desired exception on successful (dis)association
+                    pass
+            else:
+                pass  # admin role
+
+    def assign_related(self, page, name, related_set):
+        pass
+
+    def assign_related_assets(self, resource, assets):
+        for asset in assets:
+            page = self.get_by_natural_key(asset['natural_key'])
+            for name, S in asset.get('related', {}).items():
+                if name == 'roles':
+                    self.assign_roles(page, S)
+                else:
+                    self.assign_related(page, name, S)
+
+    def handle(self, client, parser):
+        if client.help:
+            parser.print_help()
+            raise SystemExit()
+
+        data = json.load(sys.stdin)
+        client.authenticate()
+        self.v2 = client.v2
+
+        for resource in data:
+            self.register_existing_assets(resource)
+
+        for resource, assets in data.items():  # FIXME: do a topological sort by dependencies
+            self.create_assets(resource, assets)
+
+        # FIXME: should we delete existing unpatched assets?
+
+        for resource, assets in data.items():
+            self.assign_related_assets(resource, assets)
+
+        return {}
+
+
+class Export(CustomCommand):
+    name = 'export'
+    help_text = 'export resources from Tower'
+
+    def extend_parser(self, parser):
+        resources = parser.add_argument_group('resources')
+
+        for resource in EXPORTABLE_RESOURCES:
+            # This parsing pattern will result in 3 different possible outcomes:
+            # 1) the resource flag is not used at all, which will result in the attr being None
+            # 2) the resource flag is used with no argument, which will result in the attr being ''
+            # 3) the resource flag is used with an argument, and the attr will be that argument's value
```

Comment:
Does this mean that it is supposed to export only the specified resource name?

Reply:
Yes.  `--users admin` should export only the user with username 'admin'.